src/parsers/manifest/metaplaylist/metaplaylist_parser.ts

```diff
@@ -293,12 +294,13 @@ function createManifest(
   }
 
   const time = performance.now();
-  let isLastPeriodKnown;
-  if (manifests.length > 0) {
-    isLastPeriodKnown = !isDynamic ||
-                        mplData.pollInterval === undefined &&
-                        manifests[manifests.length - 1].isLastPeriodKnown;
-  }
+  const isLastPeriodKnown = !isDynamic ||
+                            mplData.pollInterval === undefined &&
+                            (
+                              manifests.length === 0 ||
+                              flatMap(manifests, (m) => m.periods).length === 0 ||
```

Comment:
Why is this condition necessary?
If a sub-manifest has no period but isLastPeriodKnown set to `false` how is that different than if it has Periods?

Instead of flatmapping + length which is not very readable and not optimal, why not something like `manifests.every(m => m.periods.length === 0)`